src/java.base/share/classes/java/lang/StableValue.java

```diff
@@ -0,0 +1,747 @@
+ *
+ *     private PowerOf2Util() {}
+ *
+ *     private static final int SIZE = 6;
+ *     private static final IntFunction<Integer> ORIGINAL_POWER_OF_TWO =
+ *             v -> 1 << v;
+ *
+ *     private static final List<Integer> POWER_OF_TWO =
+ *             // @link substring="list" target="#list(int,IntFunction)" :
+ *             StableValue.list(SIZE, ORIGINAL_POWER_OF_TWO);
+ *
+ *     public static int powerOfTwo(int a) {
+ *         return POWER_OF_TWO.get(a);
+ *     }
+ * }
+ *
+ * int pwr4 = PowerOf2Util.powerOfTwo(4);   // May eventually constant fold to 16 at runtime
+ *
+ * }
+ * <p>
+ * Similarly, a <em>stable map</em> is an unmodifiable map whose keys are known at
+ * construction. The stable map values are computed when they are first accessed,
+ * using a provided {@linkplain Function}:
+ *
+ * {@snippet lang = java:
+ * class Log2Util {
+ *
+ *     private Log2Util() {}
+ *
+ *     private static final Set<Integer> KEYS =
+ *             Set.of(1, 2, 4, 8, 16, 32);
+ *     private static final UnaryOperator<Integer> LOG2_ORIGINAL =
+ *             i -> 31 - Integer.numberOfLeadingZeros(i);
+ *
+ *     private static final Map<Integer, INTEGER> LOG2 =
+ *             // @link substring="map" target="#map(Set,Function)" :
+ *             StableValue.map(CACHED_KEYS, LOG2_ORIGINAL);
+ *
+ *     public static int log2(int a) {
+ *          return LOG2.get(a);
+ *     }
+ *
+ * }
+ *
+ * int log16 = Log2Util.log2(16);   // May eventually constant fold to 4 at runtime
+ *
+ *}
+ *
+ * <h2 id="composition">Composing stable values</h2>
+ * A stable value can depend on other stable values, forming a dependency graph
+ * that can be lazily computed but where access to individual elements still can be
+ * performant. In the following example, a single {@code Foo} and a {@code Bar}
+ * instance (that is dependent on the {@code Foo} instance) are lazily created, both of
+ * which are held by stable values:
+ * {@snippet lang = java:
+ * public final class DependencyUtil {
+ *
+ *     private DependencyUtil() {}
+ *
+ *     public static class Foo {
+ *          // ...
+ *      }
+ *
+ *     public static class Bar {
+ *         public Bar(Foo foo) {
+ *              // ...
+ *         }
+ *     }
+ *
+ *     private static final Supplier<Foo> FOO = StableValue.supplier(Foo::new);
+ *     private static final Supplier<Bar> BAR = StableValue.supplier(() -> new Bar(FOO.get()));
+ *
+ *     public static Foo foo() {
+ *         return FOO.get();
+ *     }
+ *
+ *     public static Bar bar() {
+ *         return BAR.get();
+ *     }
+ *
+ * }
+ *}
+ * Calling {@code bar()} will create the {@code Bar} singleton if it is not already
+ * created. Upon such a creation, the dependent {@code Foo} will first be created if
+ * the {@code Foo} does not already exist.
+ * <p>
+ * Here is another example where a more complex dependency graph is created in which
+ * integers in the Fibonacci delta series are lazily computed:
```

Comment:
I'd probably reword this to: "Another example, which has a more complex dependency graph, is computing the Fibonacci sequence:"